### DISCUSSION TITLE
How to use page->setHeader()?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#287] [Ask a question] [answered]
Hello, 

me again. :) 
if a plugin uses the `$page->setHeader()` function, is it recommended to check if a specific HTTP header already is being set or will plugin configuration simpli override the existing headers? 

As an example I would like to set `$this->yellow->page->setHeader("Cache-Control", "no-store, no-cache, must-revalidate");` in my random page plugin to override browser cache. This is necessary to provide real randomness. The solution works fine, however I'm wondering if this could cause problems with e.g. the blog plugin which uses it's own header cache settings. My tests so far didn't, but I'm asking as a precaution. 

BTW: `$page->setHeader()` is missing in the API documentation. 

Thanks
Steffen

**Comment by markseuffert** [comment, marked as answer]
Hi Steffen,

yes you can set any HTTP header and it will override the last setting. For example core and contact plugin use the following line for pages that should not be cached by the browser `$this->yellow->page->setHeader("Cache-Control", "no-cache, must-revalidate");`. You're doing it right.

Happy coding

**Comment by schulle4u** [comment]
> You're doing it right.

Rarely enugh. :) Thanks again for explanation. 

Steffen
